nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/src/main/java/org/apache/nifi/graph/TinkerpopClientService.java

```diff
@@ -0,0 +1,538 @@
+        boolean jarsIsSet = !StringUtils.isEmpty(context.getProperty(EXTRA_RESOURCE).getValue());
+        boolean clzIsSet = !StringUtils.isEmpty(context.getProperty(EXTENSION_CLASSES).getValue());
+
+        if (jarsIsSet && clzIsSet) {
+            try {
+                final ClassLoader loader = this.getClass().getClassLoader();
+                String[] classes = context.getProperty(EXTENSION_CLASSES).evaluateAttributeExpressions().getValue().split(",[\\s]*");
+                for (String clz : classes) {
+                    Class.forName(clz, true, loader);
+                }
+            } catch (Exception ex) {
+                results.add(new ValidationResult.Builder().subject(EXTENSION_CLASSES.getDisplayName()).valid(false).explanation(ex.toString()).build());
+            }
+        }
+
+        boolean standardConfigIsSet = context.getProperty(CONTACT_POINTS).isSet()
+                && context.getProperty(PATH).isSet()
+                && context.getProperty(PORT).isSet();
+        boolean fileIsSet = context.getProperty(REMOTE_OBJECTS_FILE).isSet();
+
+        if (standardConfigIsSet && fileIsSet) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("Cannot set the configuration file and connection details properties at the same time.")
+                    .valid(false)
+                    .build());
+        }
+        if (!standardConfigIsSet && !fileIsSet) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("Connection details properties or the configuration file must be set.").valid(false).build());
+        }
+
+        if (context.getProperty(USER_NAME).isSet() && !context.getProperty(PASSWORD).isSet()) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("When specifying a username, the password must also be set").valid(false).build()
+            );
+        }
+        if (context.getProperty(PASSWORD).isSet() && !context.getProperty(USER_NAME).isSet()) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("When specifying a password, the password must also be set").valid(false).build()
+            );
+        }
+        return results;
+    }
+
+    protected Cluster.Builder setupSSL(ConfigurationContext context, Cluster.Builder builder) {
+        if (context.getProperty(SSL_CONTEXT_SERVICE).isSet()) {
+            SSLContextService service = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+            builder
+                    .enableSsl(true)
+                    .keyStore(service.getKeyStoreFile())
+                    .keyStorePassword(service.getKeyStorePassword())
+                    .keyStoreType(service.getKeyStoreType())
+                    .trustStore(service.getTrustStoreFile())
+                    .trustStorePassword(service.getTrustStorePassword());
```

Comment:
Instead of passing individual properties, the Builder supports an `sslContext()` method that accepts a Netty SslContext. The Netty JdkSslContext wraps the Java SSLContext returned from the `SSLContextService.createContext()` method.

Reply:
@exceptionfactory  -- I have updated this to utilize the `.createContext()` functionality. I am not super familiar with netty, so let me know if you see any problems with `JdkSslContext` that I have created.